pages/concepts-limit-order-book-and-matching.md

```diff
@@ -0,0 +1,86 @@
+
+2. On receiving an order:
+    - The order fails to place if it has already been cancelled.
+    - Otherwise, it is matched and/or placed on the order book, with optimistic matches stored locally.
+
+3. Validator nodes propose blocks that include all their local matches.
+
+4. When processing a new block:
+    - The node starts from the state of the prior block (the local state used to propose is ignored).
+    - The block’s changes are applied.
+    - The node replays its local state on top of the new state, during which:
+        - Cancels are preserved.
+        - Orders matched in the prior local state are re-placed.
+        - Orders may match differently, fail to place due to cancellation, or not match at all in the new state.
+
+#### Source Code References
+For further details on how the protocol handles these actions, refer to the following source code references:
+
+- See [here](https://github.com/dydxprotocol/v4-chain/blob/dc6e0a004fd81e3139a24f88b10605ab5ce16cfd/protocol/x/clob/ante/clob.go#L90) and [here](https://github.com/dydxprotocol/v4-chain/blob/2d5dfa55357abd5ead46f8baa03ed76d420849cc/protocol/x/clob/memclob/memclob.go#L103) for how the protocol reacts when (1) a cancel is seen.
+
+- When (2) [an order is placed](https://github.com/dydxprotocol/v4-chain/blob/dc6e0a004fd81e3139a24f88b10605ab5ce16cfd/protocol/x/clob/ante/clob.go#L132) and [checked to not be already cancelled](https://github.com/dydxprotocol/v4-chain/blob/749dff9cbca56eb2a6ab3a19feeb338de8db80e6/protocol/x/clob/keeper/orders.go#L780).
+
+- When (3) [proposing a block](https://github.com/dydxprotocol/v4-chain/blob/189b11217490aa5a87a4108dde0f679b0190511b/protocol/app/prepare/prepare_proposal.go#L157).
+
+- And (4) when [nodes process blocks committed by consensus](https://github.com/dydxprotocol/v4-chain/blob/4780b4cba2cab75e0af5675c3e87e551162ecf33/protocol/x/clob/abci.go#L152).
+
+## Order Messages
+
+Order instructions are limit order placements, replacements, and cancellations.
+
+> Note: This section covers short-term orders which live off-chain, in node memory, until matched.
+> 
+> Stateful orders (on-chain, consensus-speed placement) exist for longer-lived limit orders but aren't recommended for API traders.
+> More info: [Short-term vs Stateful Orders](./api_integration-trading/short_term_vs_stateful.mdx)
+
+
+### Finality and GTB
+Each limit order placement or cancellation [includes a GTB (good-til-block) field](https://github.com/dydxprotocol/v4-chain/blob/dc6e0a004fd81e3139a24f88b10605ab5ce16cfd/proto/dydxprotocol/clob/order.proto#L114-L146), which specifies the block height after which the instruction expires.
+
+While rare, it is possible for a cancel instruction to be seen by the current block proposer but not by one or more subsequent proposers. In such cases, the order could still match after the sender expects it to have been cancelled.
```

Comment:
maybe add something about how this could happen if order is not gossiped throughout the network